commit 03a28c3e27138fc18675810b81b2b499d147da84
Author: David López <not4rent@gmail.com>
Date:   Wed Mar 20 13:21:38 2019 +0100

    docker: refactor docker-compose for cleanness and readability
    
    Reorganize/split docker-compose config between multiple files for cleanness and readability
    
    Signed-off-by: David López <not4rent@gmail.com>

diff --git a/docker-compose-mysql.yml b/docker-compose-mysql.yml
new file mode 100644
index 000000000..4b286ae5f
--- /dev/null
+++ b/docker-compose-mysql.yml
@@ -0,0 +1,29 @@
+###########################################################################
+#######             FOR DEMONSTRATION PURPOSES ONLY                 #######
+###########################################################################
+#                                                                         #
+# If you have not yet read the tutorial, do so now:                       #
+#  https://www.ory.sh/docs/hydra/5min-tutorial                            #
+#                                                                         #
+# This set up is only for demonstration purposes. The login               #
+# endpoint can only be used if you follow the steps in the tutorial.      #
+#                                                                         #
+###########################################################################
+
+version: '3'
+
+services:
+
+  hydra-migrate:
+    environment:
+      - DATABASE_URL=mysql://root:secret@tcp(mysqld:3306)/mysql?parseTime=true
+
+  hydra:
+    environment:
+      - DATABASE_URL=mysql://root:secret@tcp(mysqld:3306)/mysql?parseTime=true
+    restart: unless-stopped
+
+  mysqld:
+    image: mysql:5.7
+    environment:
+      - MYSQL_ROOT_PASSWORD=secret
diff --git a/docker-compose-tracing.yml b/docker-compose-tracing.yml
new file mode 100644
index 000000000..78719edfc
--- /dev/null
+++ b/docker-compose-tracing.yml
@@ -0,0 +1,36 @@
+###########################################################################
+#######             FOR DEMONSTRATION PURPOSES ONLY                 #######
+###########################################################################
+#                                                                         #
+# If you have not yet read the tutorial, do so now:                       #
+#  https://www.ory.sh/docs/hydra/5min-tutorial                            #
+#                                                                         #
+# This set up is only for demonstration purposes. The login               #
+# endpoint can only be used if you follow the steps in the tutorial.      #
+#                                                                         #
+###########################################################################
+
+version: '3'
+
+services:
+
+  hydra:
+    depends_on:
+      - jaeger
+    environment:
+      - TRACING_PROVIDER=jaeger
+      - TRACING_PROVIDER_JAEGER_SAMPLING_SERVER_URL=http://jaeger:5778/sampling
+      - TRACING_PROVIDER_JAEGER_LOCAL_AGENT_ADDRESS=jaeger:6831
+      - TRACING_PROVIDER_JAEGER_SAMPLING_TYPE=const
+      - TRACING_PROVIDER_JAEGER_SAMPLING_VALUE=1
+
+  jaeger:
+    image: jaegertracing/all-in-one:1.7.0
+    ports:
+      - "5775:5775/udp"
+      - "6831:6831/udp"
+      - "6832:6832/udp"
+      - "5778:5778"
+      - "16686:16686"
+      - "14268:14268"
+      - "9411:9411"
\ No newline at end of file
diff --git a/docker-compose-twoc-mysql.yml b/docker-compose-twoc-mysql.yml
new file mode 100644
index 000000000..39116583a
--- /dev/null
+++ b/docker-compose-twoc-mysql.yml
@@ -0,0 +1,32 @@
+###########################################################################
+#######             FOR DEMONSTRATION PURPOSES ONLY                 #######
+###########################################################################
+#                                                                         #
+# If you have not yet read the tutorial, do so now:                       #
+# https://www.ory.sh/docs/hydra/5min-tutorial                             #
+#                                                                         #
+# This set up is only for demonstration purposes. The login               #
+# endpoint can only be used if you follow the steps in the tutorial.      #
+#                                                                         #
+###########################################################################
+
+version: '3'
+
+services:
+
+  hydra-migrate:
+    environment:
+      - DATABASE_URL=mysql://root:secret@tcp(mysqld:3306)/mysql?parseTime=true
+
+  hydra-admin:
+    environment:
+      - DATABASE_URL=mysql://root:secret@tcp(mysqld:3306)/mysql?parseTime=true
+
+  hydra:
+    environment:
+      - DATABASE_URL=mysql://root:secret@tcp(mysqld:3306)/mysql?parseTime=true
+
+  mysqld:
+    image: mysql:5.7
+    environment:
+      - MYSQL_ROOT_PASSWORD=secret
diff --git a/docker-compose-twoc.yml b/docker-compose-twoc.yml
index 53d60a6ed..5ab5f1017 100644
--- a/docker-compose-twoc.yml
+++ b/docker-compose-twoc.yml
@@ -41,8 +41,6 @@ services:
       - OAUTH2_CONSENT_URL=http://localhost:3000/consent
       - OAUTH2_LOGIN_URL=http://localhost:3000/login
       - DATABASE_URL=postgres://hydra:secret@postgresd:5432/hydra?sslmode=disable
-#     Uncomment the following line to use mysql instead.
-#      - DATABASE_URL=mysql://root:secret@tcp(mysqld:3306)/mysql?parseTime=true
       - SYSTEM_SECRET=youReallyNeedToChangeThis
       - OAUTH2_SHARE_ERROR_DEBUG=1
 #      - OAUTH2_ACCESS_TOKEN_STRATEGY=jwt
@@ -64,8 +62,6 @@ services:
       - OAUTH2_CONSENT_URL=http://localhost:3000/consent
       - OAUTH2_LOGIN_URL=http://localhost:3000/login
       - DATABASE_URL=postgres://hydra:secret@postgresd:5432/hydra?sslmode=disable
-#     Uncomment the following line to use mysql instead.
-#      - DATABASE_URL=mysql://root:secret@tcp(mysqld:3306)/mysql?parseTime=true
       - SYSTEM_SECRET=youReallyNeedToChangeThis
       - OAUTH2_SHARE_ERROR_DEBUG=1
 #      - OAUTH2_ACCESS_TOKEN_STRATEGY=jwt
@@ -86,10 +82,4 @@ services:
     environment:
       - POSTGRES_USER=hydra
       - POSTGRES_PASSWORD=secret
-      - POSTGRES_DB=hydra
-
-#  Uncomment the following section to use mysql instead.
-#  mysqld:
-#    image: mysql:5.7
-#    environment:
-#      - MYSQL_ROOT_PASSWORD=secret
+      - POSTGRES_DB=hydra
\ No newline at end of file
diff --git a/docker-compose.yml b/docker-compose.yml
index 071153856..0ca914997 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -21,8 +21,6 @@ services:
     environment:
 #      - LOG_LEVEL=debug
       - DATABASE_URL=postgres://hydra:secret@postgresd:5432/hydra?sslmode=disable
-#     Uncomment the following section to use mysql instead.
-#      - DATABASE_URL=mysql://root:secret@tcp(mysqld:3306)/mysql?parseTime=true
     command:
       migrate sql -e
     restart: on-failure
@@ -33,15 +31,10 @@ services:
       dockerfile: Dockerfile
     depends_on:
       - hydra-migrate
-#     Uncomment the following when configuring tracing
-#      - jaeger
     ports:
-      # Public port
-      - "4444:4444"
-      # Admin port
-      - "4445:4445"
-      # Port for hydra token user
-      - "5555:5555"
+      - "4444:4444" # Public port
+      - "4445:4445" # Admin port
+      - "5555:5555" # Port for hydra token user
     command:
       serve all --dangerous-force-http
     environment:
@@ -50,19 +43,11 @@ services:
       - OAUTH2_CONSENT_URL=http://localhost:3000/consent
       - OAUTH2_LOGIN_URL=http://localhost:3000/login
       - DATABASE_URL=postgres://hydra:secret@postgresd:5432/hydra?sslmode=disable
-#     Uncomment the following line to use mysql instead.
-#      - DATABASE_URL=mysql://root:secret@tcp(mysqld:3306)/mysql?parseTime=true
       - SYSTEM_SECRET=youReallyNeedToChangeThis
       - OAUTH2_SHARE_ERROR_DEBUG=1
       - OIDC_SUBJECT_TYPES_SUPPORTED=public,pairwise
       - OIDC_SUBJECT_TYPE_PAIRWISE_SALT=youReallyNeedToChangeThis
-#     - OAUTH2_ACCESS_TOKEN_STRATEGY=jwt
-#     Uncomment the following lines when configuring tracing
-#      - TRACING_PROVIDER=jaeger
-#      - TRACING_PROVIDER_JAEGER_SAMPLING_SERVER_URL=http://jaeger:5778/sampling
-#      - TRACING_PROVIDER_JAEGER_LOCAL_AGENT_ADDRESS=jaeger:6831
-#      - TRACING_PROVIDER_JAEGER_SAMPLING_TYPE=const
-#      - TRACING_PROVIDER_JAEGER_SAMPLING_VALUE=1
+#      - OAUTH2_ACCESS_TOKEN_STRATEGY=jwt
     restart: unless-stopped
 
   consent:
@@ -73,19 +58,6 @@ services:
       - "3000:3000"
     restart: unless-stopped
 
-# Uncomment the following when configuring tracing
-
-#  jaeger:
-#    image: jaegertracing/all-in-one:1.7.0
-#    ports:
-#    - "5775:5775/udp"
-#    - "6831:6831/udp"
-#    - "6832:6832/udp"
-#    - "5778:5778"
-#    - "16686:16686"
-#    - "14268:14268"
-#    - "9411:9411"
-
   postgresd:
     image: postgres:9.6
     ports:
@@ -94,9 +66,3 @@ services:
       - POSTGRES_USER=hydra
       - POSTGRES_PASSWORD=secret
       - POSTGRES_DB=hydra
-
-#  Uncomment the following section to use mysql instead.
-#  mysqld:
-#    image: mysql:5.7
-#    environment:
-#      - MYSQL_ROOT_PASSWORD=secret
diff --git a/scripts/5min-tutorial.sh b/scripts/5min-tutorial.sh
new file mode 100755
index 000000000..1002b8a77
--- /dev/null
+++ b/scripts/5min-tutorial.sh
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+DB=${DB:-postgres}
+TRACING=${TRACING:-false}
+TWOC=${TWOC:-false}
+
+DC="docker-compose -f docker-compose.yml"
+if [[ $DB == "mysql" ]]; then
+    DC+=" -f docker-compose-mysql.yml"
+fi
+if [[ $TRACING == true ]]; then
+    DC+=" -f docker-compose-tracing.yml"
+fi
+if [[ $TWOC == true ]]; then
+    DC+=" -f docker-compose-twoc.yml"
+fi
+DC+=" up --build"
+
+$DC
+
